numpy/lib/nanfunctions.py

```diff
@@ -546,6 +548,73 @@ def nanargmax(a, axis=None):
     return res
 
 
+def _nanptp_dispatcher(a, axis=None):
+    return (a,)
+
+
+@array_function_dispatch(_nanptp_dispatcher)
+def nanptp(a, axis=None):
+    """
+    Range of values (maximum - minimum) along an axis, ignoring any NaNs.
+    When slices with less than two non-NaN values are encountered,
+    a ``RuntimeWarning`` is raised and Nan is returned for that slice.
+
+    Parameters
+    ----------
+    a : array_like
+        Array containing numbers whose peak-to-peak is desired. If `a` is not an
+        array, a conversion is attempted.
+    axis : {int, tuple of int, None}, optional
+        Axis or axes along which the minimum is computed. The default is to compute
+        the minimum of the flattened array.
+
+    Returns
+    -------
+    nanptp : ndarray
+        An array with the same shape as `a`, with the specified axis
+        removed.  If `a` is a 0-d array, or if axis is None, np.nan is
+        returned, and a warning raised.  The same dtype as `a` is returned.
+
+    Notes
+    -----
+    NumPy uses the IEEE Standard for Binary Floating-Point for Arithmetic
+    (IEEE 754). This means that Not a Number is not equivalent to infinity.
+    Positive infinity is treated as a very large number and negative
+    infinity is treated as a very small (i.e. negative) number.
+
+    If the input has a integer type the function is equivalent to np.ptp.
+
+    Examples
+    --------
+    >>> a = np.array([[1, 2], [3, np.nan], [-3, 1]])
+    >>> np.nanptp(a)
+    6.0
+    >>> np.nanptp(a, axis=0)
+    array([6., 1.])
+
+    The command below will raise a RuntimeWarning
+    
+    >>> np.nanptp(a, axis=1)
+    array([ 1., nan, 4.])
+
+    When positive infinity and/or negative infinity are present:
+
+    >>> np.nanptp([1, 2, np.nan, np.inf])
+    inf
+    >>> np.nanptp([1, 2, np.nan, np.NINF])
+    inf
+
+    """
+    res = nanmax(a, axis=axis) - nanmin(a, axis=axis)
+    valid_mask = ~np.isnan(a)
```

Comment:
Is this necessary? Won't `nanmax` and `nanmin` already have each given a warning for all-nan axes?

Reply:
Looks like it isn't, I misunderstood the expected behavior in this case. I thought you need at least two values to have a valid peak-to-peak answer, but it looks like a single value should return `0`. I will fix this, thanks for catching it. Are the warnings from `nanmin` and `nanmax` acceptable, or do we want the function to have its own warning?